packages/babel-plugin-strip-runtime/src/__tests__/jsx-pragma.test.ts

```diff
@@ -0,0 +1,370 @@
+// @compiled/babel-plugin and @compiled/babel-plugin-strip-runtime
+// and how they interact with:
+//
+// * JSX pragma comments /** @jsx jsx */ and
+//     /** @jsxImportSource @compiled/react */, and
+// * @babel/preset-react's JSX pragma-related config options
+//     (discussed in
+//     https://babeljs.io/docs/babel-plugin-transform-react-jsx
+//     and https://babeljs.io/docs/babel-preset-react)
+
+describe('Compiled setup with classic runtime', () => {
+  describe('with JSX pragma', () => {
+    describe('if only Compiled is used in file', () => {
+      it('converts JSX elements to React.createElement', () => {
+        const code = `
+        /** @jsx jsx */
+        import { css, jsx } from '@compiled/react';
+
+        const Component = () => (
+          <div css={{ fontSize: 12, color: 'blue' }}>
+            hello world 2
+          </div>
+        );
+
+        const Component2 = () => (
+          <div css={css({ fontSize: 12, color: 'pink' })}>
+            hello world 2
+          </div>
+        );
+      `;
+
+        const codeWithRenamedImport = `
+        /** @jsx myJsx */
+        import { css, jsx as myJsx } from '@compiled/react';
+
+        const Component = () => (
+          <div css={{ fontSize: 12, color: 'blue' }}>
+            hello world 2
+          </div>
+        );
+
+        const Component2 = () => (
+          <div css={css({ fontSize: 12, color: 'pink' })}>
+            hello world 2
+          </div>
+        );
+      `;
+
+        for (const c of [code, codeWithRenamedImport]) {
+          const actual = transform(c, {
+            run: 'both',
+            runtime: 'classic',
+          });
+
+          expect(actual).toContain("import * as React from 'react';");
+          // All components should look like this:
+          //     const Component = () => /*#__PURE__*/ React.createElement(...)
+          expect(actual).toMatch(
+            /Component = \(\) =>\n(\s)*\/\*#__PURE__\*\/ React.createElement/m
+          );
+          expect(actual).toMatch(
+            /Component2 = \(\) =>\n(\s)*\/\*#__PURE__\*\/ React.createElement/m
+          );
+          // All traces of JSX pragma or JSX imports are removed
+          expect(actual).not.toContain('jsx');
+        }
+      });
+    });
+
+    describe('if only Emotion is used in file', () => {
+      it('does not process file', () => {
+        const code = `
+          /** @jsx jsx */
+          import { css, jsx } from '@emotion/react';
+
+          const Component = () => (
+            <div css={{ fontSize: 12, color: 'blue' }}>
+              hello world 2
+            </div>
+          );
+
+          const Component2 = () => (
+            <div css={css({ fontSize: 12, color: 'pink' })}>
+              hello world 2
+            </div>
+          );
+        `;
+
+        const actual = transform(code, {
+          run: 'both',
+          runtime: 'classic',
+        });
+
+        expect(actual).toMatchInlineSnapshot(`
```

Comment:
Instead of a snapshot can we do a exact match?

```suggestion
        expect(actual).toEqual(code);
```

Reply:
`actual` isn't equal to `code` though 🤔

Reply:
Does this mean it is being transformed? 😅

Reply:
yes but not by Compiled, but rather by `@babel/preset-react`

i'll update the wording of the test name